packages/admin/cms-admin/src/pages/pagesPage/PagesPage.tsx

```diff
@@ -202,7 +203,11 @@ export function PagesPage({
 
                         const EditComponent = documentType.editComponent;
 
-                        return EditComponent ? <EditComponent id={selectedId} category={category} /> : null;
+                        return EditComponent ? (
+                            <LocalPageTreeNodeDocumentAnchorsProvider>
```

Comment:
can't you create that context in createUsePage (and avoid duplicating state)?

Reply:
createUsePage currently doesn't provide a React component, but a React hook. I could wrap the context around each root block's adminUI, which is used in the edit component (in application code)

```tsx
<AdminComponentRoot
    title={intl.formatMessage({ id: "cometDemo.pages.pages.page.edit.pageBlocks.title", defaultMessage: "Page" })}
>
   {rootBlocksApi.content.adminUI} // Context provider is in here
</AdminComponentRoot>
```

Reply:
Done in 123cc6a9560e45ebd6349dfa53abd7fda318584b